pagopa/io-app
# chore: [IOBP-163] Handle 401 status code in IDPay flows (Fast Login)

## Short description
This PR adds the handling of 401 status code inside IDPay flows. Specifically, whenever a 401 status code is received, the IDPay flows will start an authentication flow provided by the Fast Login sagas.
**Note**: currently XState machine does not support the preserving of the state after a 401 status code. This will handled in a later PR (IOBP-164).

## List of changes proposed in this pull request
- Added `withRefreshApiCall` in IDPay sagas
- Added session handling in IDPay XState machines

## How to test
With the `io-dev-server` you should firs enable Fast Login feature flag in `backend.ts` like the example below:

```js
    ...
    fastLogin: {
      min_app_version: {
        ios: "1.2.3",
        android: "1.2.3"
      }
    },
    ...
```
Then navigate any IDPay sections and check that every flow is working. You should expect a 401 status code every 60s.
